torch/lib/THD/master_worker/master/Master.cpp

```diff
@@ -4,29 +4,46 @@
 #include "../worker/Worker.h"
 #include "../../process_group/General.hpp"
 
+#include <string>
+#include <thread>
 
 namespace thd {
-
 namespace master {
 
 std::unique_ptr<MasterCommandChannel> masterCommandChannel;
+std::thread masterErrorThread;
 
-} // namespace master
+void errorHandler() {
+  for (;;) {
+    auto error = masterCommandChannel->recvError();
+    THDState::s_error = "Error (worker " + std::to_string(std::get<0>(error)) + "): " + std::get<1>(error);
+  }
+}
 
+} // namespace master
 } // namespace thd
 
 using namespace thd;
+using namespace thd::master;
 
 bool THDMasterWorkerInit(THDChannelType channel_type) {
   if (!THDProcessGroupInit(channel_type)) return false;
 
   if (dataChannel->getRank() > 0) {
-    THDWorkerMain();
+    // Worker initialization. Can fail at start but then goes into inifity loop.
+    return THDWorkerMain();
+  }
+
+  // Master initilization
+  masterCommandChannel.reset(new MasterCommandChannel());
+  bool ok = masterCommandChannel->init();
```

Comment:
Is there a reason for this `ok` variable? (You don't use this structure in other files, for example:  torch/lib/THD/master_worker/worker/Worker.cpp)